js/google_ima.js

```diff
@@ -536,10 +559,14 @@ require("../html5-common/js/utils/utils.js");
           //if we are trying to play an linear ad then we need to request the ad now.
           if (this.currentAMCAdPod.ad.forced_ad_type != _amc.ADTYPE.NONLINEAR_OVERLAY)
           {
-            //reset adRequested and adTagUrl so we can request another ad
-            _resetAdsState();
-            this.adTagUrl = this.currentAMCAdPod.ad.tag_url;
-            _trySetupAdsRequest();
+            //if this is the preroll we preloaded then don't try to request it again.
+            if (!(this.preloadAds && this.numAdsPlayed === 1 && this.currentAMCAdPod.position === 0))
```

Comment:
I can't sneak anything past you :).  With ad preloading, we only preloads the first preroll if it exists.  So this is checking if the first ad to play is a preroll and we are preloading, then we have already requested the ad, so skip requesting it again.

Now that I've explained it, I probably need to go back and figure out a more generic way to mark ads in the AMC as being preloaded and then we just check a boolean.  Then we can reuse this for anything.